### PR TITLE
Bridge gap between peaks and react

### LINKED ISSUE #27: Explore filling the gap between Peaks.js and React app
### Description
The React app is unable to access some events triggered (such as changing segments by dragging handles on the waveform) by Peaks.js directly. In order to fill this gap we can explore the following options;
1. Writing a JavaScript wrapper to bridge the gap
2. Implement a React component to contain the Peaks instance
3. Bind Peaks events directly to redux actions and reducers and subscribe components in the app to them, to get updates from Peaks

child of samvera-labs/avalon-bundle#231

### PR DESCRIPTION
**Pull request by Dananji**
@adamjarling This PR contains the solution for the problem of accessing Peaks events triggered when editing timespans by dragging handles in the Peaks waveform.
The uses is implemented using, `rxjs` library to access the Peaks event `segments.dragged` and react middleware to dispatch an action to update the state whenever the `segments.dragged` event fires.
In this version, we can edit a timespan from both the inline edit form and Peaks waveform.
This fixes #27

### CODE REVIEW

**Comment by Dananji** [comment]
@adamjarling  Once we merge the existing PR for #28, I will rebase this branch on `master` and resolve the merge conflicts.

File: src/api/Utils.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ export const BASE_URL = useLocalData
   : 'https://spruce.dlib.indiana.edu';
 
 // Masterfile ID on the server
-export const masterFileID = 'j3860704z';
+export const masterFileID = 'd791sg30j';
```
Comment (Dananji) [inline src/api/Utils.js]: @adamjarling I ingested a new file and this is the new masterfile ID. Earlier one was deleted when cleaning up the server.

**Comment by adamjarling** [comment]
Sounds good.  Ping me when #28 is merged and the rebase is complete.

**Comment by Dananji** [comment]
@adamjarling When I rebase the functionality of updating the time from both Peaks and input field breaks, which worked before.

**Comment by Dananji** [comment]
@adamjarling I rebased this branch. You can check it out now.

**Comment by adamjarling** [comment]
This looks like really good work.  Nice job.  The Git commit history looks like it's doubling up on commits and/or messaging, guessing this happened with the rebase?   We should aim to keep this as clean as possible moving forward, so we can inspect each commit if need be to troubleshoot.  It's fine for now...I looked into it a bit and not sure what's going on, so as long as the code is working properly, we can just let it be.

Typically what I'll do when working on a larger branch, over the course of a few days with multiple commits, is keep rebasing on the `master` branch periodically, to make sure my own branch code is not getting too far off of where `master` is going.  Ideally there shouldn't be a commit dedicated to a rebase, unless it needs it for some reason, maybe a merge conflict or something.  Rebasing should just be part of the Git flow for a repo.  

All in all, this looks really good though.   It will take me more time to get in and work on some issues to fully understand the ins and outs of what it accomplishes, but everything runs locally for me, and the code itself is clean and well organized.

**Comment by adamjarling** [comment]
Another Git technique that I've found useful when things start getting complicated with Git history and things look odd, is to create a new, clean branch from master, and then cherry-pick my local commits, in the order I want them, to create a clean Git history when doing a PR back into `master`.

**Comment by Dananji** [comment]
@adamjarling Yeah this happened with the rebase. Oh... yes rebasing on master frequently while developing would have make this PR more clear. Thanks for the tip :+1:  I will keep that in mind when I do something for a longer time in future :)

Yes, there are some addition, which might take some time to fully comprehend (even for me) :) I have used redux middleware and rxjs observables to handle asynchronous events from Peaks and create an observable sequence from the Peaks events respectively.

**Pull request merged**